chore: metrics collection job should also run on develop

vaind opened the pull request:
The previous PR that introduces the Replay SDK metrics job already contained code that collected baseline info from the main branch.   
However, after making a change to only run when a specific label is added, this feature got lost... 
This PR should fix that.

mydea commented on .github/workflows/build.yml:
```diff
@@ -795,7 +795,7 @@ jobs:
     runs-on: ubuntu-20.04
     timeout-minutes: 30
-    if: contains(github.event.pull_request.labels.*.name, 'ci-overhead-measurements')
+    if: contains(github.event.pull_request.labels.*.name, 'ci-overhead-measurements') || github.ref == 'refs/heads/master'
```
You can use `needs.job_get_metadata.outputs.is_master == 'true'` here instead!

vaind replied:
ok, changed

mydea commented:
In the meanwhile we changed the main branch to develop (I updated the PR target). please rebase this on latest develop, and change `is_master` to `is_develop` :)

vaind commented:
> In the meanwhile we changed the main branch to develop (I updated the PR target). please rebase this on latest develop, and change `is_master` to `is_develop` :)

done
